edb/schema/utils.py

```diff
@@ -658,3 +659,27 @@ def _intersection_error(schema: s_schema.Schema,
         -> errors.SchemaError:
     names = ', '.join(sorted(c.get_displayname(schema) for c in components))
     return errors.SchemaError(f'cannot create an intersection of {names}')
+
+
+MAX_INT64 = 2 ** 63 - 1
+MIN_INT64 = -2 ** 63
+
+
+def const_ast_from_python(val: Any) -> qlast.BaseConstant:
+    if isinstance(val, str):
+        return qlast.StringConstant.from_python(val)
+    elif isinstance(val, bool):
+        return qlast.BooleanConstant(value='true' if val else 'false')
+    elif isinstance(val, int):
+        if MIN_INT64 <= val <= MAX_INT64:
+            return qlast.IntegerConstant(value=str(val))
+        else:
+            return qlast.BigintConstant(value=f'{val}n')
```

Comment:
I think we should raise an error in this case.

Reply:
OK, I've put a `ValueError` on that branch.

The GraphQL translator has its own ifs covering the int64/bigint case.